workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfiguration.cs

```diff
@@ -6,138 +6,94 @@
 
 namespace Improbable.Gdk.BuildSystem.Configuration
 {
-    [CreateAssetMenu(fileName = "SpatialOS Build Configuration", menuName = CreateMenuPath)]
-    public class SpatialOSBuildConfiguration : SingletonScriptableObject<SpatialOSBuildConfiguration>
-    {
-        internal const string CreateMenuPath = "SpatialOS/Build Configuration";
-
+    [CreateAssetMenu(fileName = "SpatialOS Build Configuration", 
+        menuName = EditorConfig.ParentMenu + "/" + EditorConfig.BuildConfigurationMenu)]
+    public class SpatialOSBuildConfiguration : ScriptableSingleton<SpatialOSBuildConfiguration>
+    {        
+        [SerializeField] public List<WorkerBuildConfiguration> WorkerBuildConfigurations = 
+            new List<WorkerBuildConfiguration>();
+        
         [SerializeField] private bool isInitialised;
+        
+        public BuildEnvironmentConfig GetEnvironmentConfigForWorker(string workerType, BuildEnvironment environment)
+        {
+            var config = WorkerBuildConfigurations.FirstOrDefault(x => x.WorkerType == workerType);
+            if (config == null)
+            {
+                throw new ArgumentException($"Unknown worker type {workerType}.");
+            }
 
-        [SerializeField] public List<WorkerBuildConfiguration> WorkerBuildConfigurations;
+            return config.GetEnvironmentConfig(environment);
+        }
 
-        public override void OnEnable()
+        internal string[] GetScenePathsForWorker(string workerType)
+        {
+            return GetScenesForWorker(workerType)
+                .Where(sceneAsset => sceneAsset != null)
+                .Select(AssetDatabase.GetAssetPath)
```

Comment:
Is there a possibility where `sceneAsset != null`, but `AssetDatabase.GetAssetPath(sceneAsset)` returns null?